src/components/ValidatedTextInput.tsx

```diff
@@ -27,32 +27,16 @@ export interface PhoneValidatorProps {
   countryCallingCode: string
 }
 
-export interface IntegerValidatorProps {
-  validator: ValidatorKind.Integer
-}
-
-export interface DecimalValidatorProps {
-  validator: ValidatorKind.Decimal
-  numberOfDecimals: number
-}
-
-// Required props for a custom input validator
-export interface CustomValidatorProps {
-  validator: ValidatorKind.Custom
-  customValidator: (input: string) => string
-}
-
-export type ValidatorProps =
-  | PhoneValidatorProps
-  | IntegerValidatorProps
-  | DecimalValidatorProps
-  | CustomValidatorProps
-
-export type ValidatedTextInputProps = OwnProps & ValidatorProps & TextInputProps
+export type ValidatedTextInputProps = OwnProps & PhoneValidatorProps & TextInputProps
 
 export default class ValidatedTextInput extends React.Component<ValidatedTextInputProps> {
   onChangeText = (input: string): void => {
-    const validated = validateInput(input, this.props)
+    // some countries support multiple phone number lengths. since we add the
+    // country code to a validated phone number in the UI, we need to omit the
+    // country code from subsequent validations if the user types a longer number
+    const rawInput = input.split(this.props.countryCallingCode)[1] ?? input
```

Comment:
the value of `this.props.countryCallingCode` is something like `+82`

Reply:
some qs:
- why do we add the country code to the txt field?
- can the country code appear in the number `+82 10-1282-1234` -> `['+', ' 10-12', '-1234']`?

Reply:
we add the country code to the text field after the user's input is validated. so entering a dutch phone number, i will see something like "061912345" in the text field until i enter the last digit, then the input will change to "+31 619-123-456". this is not a technical constraint, i think it's a product requirement - basically this update to show the full phone number is the cause of all our problems. fwiw i do think it's nice to show the full phone number with the country code, as it's not clear from the UI what the country code is otherwise.

i don't see a reason to make the data appear as `['+', ' 10-12', '-1234']` - i think you are suggesting that we use a separate value for the data rather than the display value, but we already do this further up the chain of components. we keep a nice object that describes the display number / country code / e164number, but this component is the lowest level display component, that responds to the user input. so to separate the displayed value from the input to this onChange handler would require an extra layer of state, i'm not sure it's worth the extra complexity?

Reply:
Initially we were showing +XX next to the flag,
And only showing the local format in the text input.
I forgot why we switched though. Maybe some users were still entering +XX (because pasting?) or auto-complete from the OS.